README.md

```diff
@@ -1,31 +1,34 @@
 
-**Note** This project is currently only supported for `nebula` boards and does not provide support for `galaxy` boards.
+**Note:** Currently only Tenstorrent `nebula` boards are supported and `galaxy` boards are not yet supported.
 
 ## Build Process
-tt-xla and stableHLO integration with tt-mlir compiler is still under progress. This build flow provides an easy way to experiment with tt-xla, stableHLO, and tt-mlir infrastructure. This build process will be updated to enhance the user experience. Before compiling, please ensure environtment variable `TTMLIR_TOOLCHAIN_DIR` is set.
+tt-xla integration with tt-mlir compiler is still under progress and currently it depends on tt-mlir toolchain for build. This build flow provides an easy way to experiment with tt-xla, StableHLO and tt-mlir infrastructure. Build process will be updated in the future to enhance the user experience.
 
-### tt-mlir environment
-- clone tt-mlir [repo](https://github.com/tenstorrent/tt-mlir).
-- Follow tt-mlir build [instructions](https://docs.tenstorrent.com/tt-mlir/build.html) to build tt-mlir environment and install all dependencies.
+### tt-mlir toolchain
+Before compiling tt-xla, tt-mlir toolchain needs to be built:
+- Clone [tt-mlir](https://github.com/tenstorrent/tt-mlir) repo
+- Follow tt-mlir [build instructions](https://docs.tenstorrent.com/tt-mlir/build.html) to setup the environment and build the toolchain
 
 ### tt-xla
-This `tt-xla` repo is updated to use cmake instead of bazel and made compatible with tt-mlir compiler.
-```
+Before running these commands to build tt-xla, please ensure that the environtment variable `TTMLIR_TOOLCHAIN_DIR` is set to point to the tt-mlir toolchain directory created above as part of tt-mlir environment setup (for example `export TTMLIR_TOOLCHAIN_DIR=/opt/ttmlir-toolchain/`). You can also set `export LOGGER_LEVEL=DEBUG` in order to enable debug logs.
+
+```bash
 git clone git@github.com:tenstorrent/tt-xla.git
 cd tt-xla
 source venv/activate
-cmake -G Ninja -B build
+cmake -G Ninja -B build # -DCMAKE_BUILD_TYPE=Debug in case you want debug build
 cmake --build build
 ```
 
 ## Testing
-tt-xla repo contains various tests in `tests` directory. To run them, please run `pytest -v tests` from project root directory.
+tt-xla repo contains various tests in `tests` directory. To run them all, please run `pytest -v tests` from project root directory. To run individual test, `pytest -svv` is recommended in order to capture all potential error messages down the line.
```

Comment:
Currently (but waiting on [PR review](https://github.com/tenstorrent/tt-xla/pull/100/files) to see what the final shape of test infra will be) both tests and test infra is located in `tests` folder. 

Might be better if we nested tests themselves in `tests/unit_tests` or provide `tests/run_unit_tests.sh` script so that `pytest` is run on some top level folder which contains only tests themselves like `pytest -svv tests/unit_tests`. 

Or provide `tests/run_unit_tests.sh` which can circumvent that by explicitly targeting folders.

Reply:
I thought pytest will ignore files not containing tests? Anyways, let's discuss this separately and then you can update the readme if needed with that PR.